### DISCUSSION TITLE
The node.head return the HGL of the node?

### DISCUSSION THREAD

**Discussion by CChaoE17** [pyswmm/pyswmm#470] [Q&A] [answered]
Hi,
I am trying to use the pyswmm to run the inp file and get the HGL of each node after the simulation. However, when I used the node.head in the nodes module, I found the returning values of normal junction nodes were not the HGL values but some values that were less than the HGL value in the .rpt file (67, 76.5, 78.5, and 76 for j1, j2, j3, and j4 in the rpt file), while for the storage nodes, the returning head values are equal to the HGL values in the rpt file. So, I am wondering if the node.head returns the HGL value of the node? If yes, then why there is a difference between normal nodes and storage nodes. 
Any response will be appreciated! Thanks so much!
The following lines are my codes and results:
with Simulation(r'case2.inp') as sim:
    for step in enumerate(sim):
        pass
    j1 = Nodes(sim)["N25_node"]
    j2 = Nodes(sim)["N35_Node"]
    j3 = Nodes(sim)["N45_Node"]
    j4 = Nodes(sim)["N105_Node"]
    j5 = Nodes(sim)["N05_Storage"]
    j6 = Nodes(sim)["N15_Storage"]
    j7 = Nodes(sim)["N55_Storage"]
    j8 = Nodes(sim)["N65_Storage"]
    j9 = Nodes(sim)["N75_Storage"]
    j10 = Nodes(sim)["N85_Storage"]
    j11 = Nodes(sim)["N95_Storage"]
    # print(j1.invert_elevation)
    # print(j1.statistics["max_depth"])
    print(j1.head)
    # print(j2.invert_elevation)
    # print(j2.statistics["max_depth"])
    print(j2.head)
    print(j3.head)
    print(j4.head)
    print(j5.head)
    print(j6.head)
    print(j7.head)
    print(j8.head)
    print(j9.head)
    print(j10.head)
    print(j11.head)
63.01749607691447
73.50953993266458
76.0095082393627
72.02155704488484
30.0
34.0
33.0
38.0
37.0
38.0
49.0

**Comment by karosc** [comment]
@CChaoE17 , to which section in the RPT file are you comparing the pyswmm values?

The HGL in the RPT file is usally the Maximum HGL. 
![image](https://github.com/pyswmm/pyswmm/assets/36245370/041ee0d7-2f20-47da-b45e-58d7f3f02e9b)


The pyswmm code you have above returns the HGL at the last timestep of the simulation.

You should be able to get to the value in the RPT file by adding the max_depth statistic to the node invert elevations a la:

```python
j1.statistics["max_depth"] + j1.invert_elevation
```

Also, I am converting this to a discussion since it's likely not a bug.

**Reply by CChaoE17** [reply]
Hi, [karosc](https://github.com/karosc). Thank you for the reply.
I compared the results with the rpt file obtained by running the model using pyswmm. The first line of rpt file is "OWA STORM WATER MANAGEMENT MODEL - VERSION v5.2.3 (OWA 2023-06-06)". This is the screenshot all the nodes. 
![image](https://github.com/pyswmm/pyswmm/assets/56448179/7edd5a73-def0-4c0f-821c-ad0d2b513ec8)
But if you mean the water head in my code only returns the value of the last minute of a running process, and if that's why the value was different from the maximum HGL in the rpt file, that would make much more sense to the difference I found. Anyway, thank you very much!

**Reply by karosc** [reply, marked as answer]
Yes, running the pyswmm code below would print the HGL of j2 at the final timestep in the simulation:

```python
for step in enumerate(sim):
	pass
print(j2.head)
```

The rpt final contains the maximum simulated HGL value during the entire simulation. That's probably why they are different. I showed code above that should allow you to extract a maximum value statistic using pyswmm.
